spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/RetryLoadBalancerInterceptor.java

```diff
@@ -113,10 +143,28 @@ private RetryTemplate createRetryTemplate(String serviceName, HttpRequest reques
 		if (retryListeners != null && retryListeners.length != 0) {
 			template.setListeners(retryListeners);
 		}
-		template.setRetryPolicy(!this.lbProperties.isEnabled() || retryPolicy == null
+		template.setRetryPolicy(!this.retryProperties.isEnabled() || retryPolicy == null
 				? new NeverRetryPolicy() : new InterceptorRetryPolicy(request,
 						retryPolicy, this.loadBalancer, serviceName));
 		return template;
 	}
 
+	private Set<LoadBalancerLifecycle> supportedLifecycleProcessors(String serviceId) {
```

Comment:
I suspect this will be a common thing to do. Maybe move to a common class after this PR is merged.

Reply:
Will get this done now already.

Reply:
I am adding a static method to avoid instantiating new objects at every filter/ load-balancing calls since we only know all the data to filter against at method level and would not be able to instantiate it with appropriate field values at a singleton field initialisation level.